sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -347,9 +347,16 @@ def get_all_options(
           else:
             parser.add_argument(split[0], type=str)
           i += 1
-        else:
+        elif unknown_args[i].startswith('--'):
           parser.add_argument(unknown_args[i], type=str)
           i += 2
+        else:
+          # skip all binary flags used with '-' and not '--'.
+          # ex: using -f instead of --f (or --flexrs_goal) will prevent
+          # argument validation before job submission and can be incorrectly
```

Comment:
shall we log these as warning? @tvalentyn

Reply:
Added warning log

Reply:
somehow we are not getting these warnings.